From b7781823987a6ee5d891a24da4209e6c3a732bc1 Mon Sep 17 00:00:00 2001
From: Gu Zitao <guzitao@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:06 +0800
Subject: [PATCH] sw64: fix the VDSO symbol generation for nm
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56P0Z

--------------------------------

We were relying on GNU nm's ability to extract our VDSO symbols, but
newer nm will always display version strings after symbol names, which
cause so2s.sh to fail to generate vdso-syms.S. Fix so2s.sh to make it
work.

Besides, remove the copyright strings from so2s.sh.

Signed-off-by: Gu Zitao <guzitao@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/vdso/so2s.sh | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/sw_64/kernel/vdso/so2s.sh b/arch/sw_64/kernel/vdso/so2s.sh
index 8f23ac544d1b..e1763af8e730 100755
--- a/arch/sw_64/kernel/vdso/so2s.sh
+++ b/arch/sw_64/kernel/vdso/so2s.sh
@@ -1,5 +1,4 @@
 #!/bin/sh
 # SPDX-License-Identifier: GPL-2.0+
-# Copyright 2020 Palmer Dabbelt <palmerdabbelt@google.com>
 
-grep -v "LINUX" | sed 's/\([0-9a-f]*\) T \([a-z0-9_]*\)/.globl\t\2\n\2:\n.quad\t0x\1/'
+grep "__vdso_" | sed 's/\([0-9a-f]*\) T \([a-z0-9_]*\)\(@@LINUX_.*\)*/.globl\t\2\n\2:\n.quad\t0x\1/'
-- 
2.34.1

